File: dbflow/src/androidTest/java/com/raizlabs/android/dbflow/test/structure/join/CompanyDepartmentJoin.java
package com.raizlabs.android.dbflow.test.structure.join;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.QueryModel;
import com.raizlabs.android.dbflow.structure.BaseQueryModel;
import com.raizlabs.android.dbflow.test.TestDatabase;

/**
 * Description: Represents the join of two tables of {@link Company} and {@link Department}.
 */
@QueryModel(database = TestDatabase.class)
public class CompanyDepartmentJoin extends BaseQueryModel {

    @Column
    String emp_id;

    @Column
    String name;

    @Column
    String dept;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/join/CompanyDepartmentJoin.kt
package com.raizlabs.android.dbflow.test.structure.join

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.QueryModel
import com.raizlabs.android.dbflow.structure.BaseQueryModel
import com.raizlabs.android.dbflow.test.TestDatabase

/**
 * Description: Represents the join of two tables of [Company] and [Department].
 */
@QueryModel(database = TestDatabase::class)
class CompanyDepartmentJoin : BaseQueryModel() {

    @Column
    var emp_id: Long = 0

    @Column
    var name: String? = null

    @Column
    var dept: String? = null
}
